apiary-receivers/apiary-receiver-common/src/main/java/com/expedia/apiary/extensions/receiver/common/messaging/MessageEvent.java

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright (C) 2018-2019 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expedia.apiary.extensions.receiver.common.messaging;
+
+import java.util.Map;
+
+import com.expedia.apiary.extensions.receiver.common.event.ListenerEvent;
+
+public class MessageEvent {
+  private ListenerEvent event;
+  private Map<MessageProperty, String> messageProperties;
```

Comment:
Thanks for the thoughts. The `MessageEvent` class is pretty open at the moment for extending as it isn't `SQS` specific. So in theory to add more receivers, if the new receivers needed to return any extra properties, we would add them as an Enum which implements the marker interface `MessageProperty` (this last bit is from my recent commit).

If we added a `getMessageReceiptHandle()` to this class then it would tie it to SQS when at the moment its a common class.

What do you think?

Reply:
Right. Ok. makes sense.